pkg/controllers/machinesetsync/machineset_sync_controller_test.go

```diff
@@ -370,7 +454,7 @@ var _ = Describe("With a running MachineSetSync controller", func() {
 
 					It("should not populate the field", func() {
 						Eventually(k.Object(mapiMachineSet), timeout).Should(
-							HaveField("Finalizers", BeEmpty()),
+							HaveField("Finalizers", Not(ContainElements("foo", "bar"))),
```

Comment:
Foo bar?

Reply:
Further up we populate `foo`:`bar`, and then expect no finalizers to be propagated. However, once we add deletion logic we rely on sync finalizers being in place.

 This keeps the meaning of the existing test :)